### PR TITLE
Improvements to installation docs

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
Docs

### Description
This PR improves the manual installation guide by using more styling features to make it more readable.

It also migrates the container docs to the main documentation and removes the old migration documentation.

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
https://github.com/hedgedoc/hedgedoc/issues/721
https://github.com/hedgedoc/container/issues/160

### CODE REVIEW

**Comment by davidmehren** [comment]
The Docker docs that I migrated from the container repository still have some issues:
- The "Upgrading to 1.7" instructions should be included in our main changelog. We could either add them to the 1.7 release notes on GitHub or get our [releases page](https://hedgedoc.org/releases/) to display additional release notes (cc @DerMolly).
- The backup, restore and upgrade instructions are only for PostgreSQL.
- I don't think the "custom build" instructions are very helpful, so I would like to remove them or move them to a separate development documentation.

**Comment by DerMolly** [comment]
I think the change to release page of the website depends on what the final decision in https://github.com/hedgedoc/hedgedoc/issues/721 is?

**Comment by davidmehren** [comment]
> I think the change to release page of the website depends on what the final decision in #721 is?

The open question in #721 is whether we want to remove the release notes from the application, which is pretty orthogonal to "display a unified changelog on our releases page"

File: docs/content/setup/manual-setup.md [hunk 1]
```diff
@@ -61,3 +62,64 @@ If you are upgrading HedgeDoc from an older version, follow these steps:
...
+    - Make sure `ReadWritePaths` contains all directories HedgeDoc might write to. This may
+    include the `public/uploads` folder if you configure local storage. If you use SQLite, you
+    must also include the directory where the database file is saved. **Do not save the SQLite
+    file in the root directory of the HedgeDoc installation**, but create a subfolder like `db`!
+
+```ini
+[Unit]
+Description=HedgeDoc - The best platform to write and share markdown.
+Documentation=https://docs.hedgedoc.org/
+After=network.target
```
Comment (InnayTool) [inline docs/content/setup/manual-setup.md:85]: Shouldn't we add the database server here?

Comment (davidmehren) [reply docs/content/setup/manual-setup.md:87]: added

**Comment by DerMolly** [comment]
> > I think the change to release page of the website depends on what the final decision in #721 is?
> 
> The open question in #721 is whether we want to remove the release notes from the application, which is pretty orthogonal to "display a unified changelog on our releases page"

Well, if you merge all the changelogs, then the "Upgrading to 1.7" instructions would be merged, too, wouldn't it? If that is not the case or we end up not doing this one would need to add the ability to display additional release notes.

**Comment by davidmehren** [comment]
> Well, if you merge all the changelogs, then the "Upgrading to 1.7" instructions would be merged, too, wouldn't it? If that is not the case or we end up not doing this one would need to add the ability to display additional release notes.

There was no one against merging the changelogs in #721, so I consider that decision done. As I said, the remaining discussion is concerned with not shipping the release notes to users.

The question now is really just the one that I mentioned: We need to decide if we want to include the docker changelog into the release notes of hedgedoc/hedgedoc or if we keep that separate and merge it only on our releases page.

And yes, the "Upgrading to 1.7" section would then be removed from this PR, there is a reason why this is still a Draft ;)

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — DISMISSED** [review]

**Comment by davidmehren** [comment]
This is now deployed to https://hedgedoc-docs-staging.web.app

File: docs/content/setup/manual-setup.md [hunk 2]
```diff
@@ -1,41 +1,139 @@
...
-1. Check if you meet the [requirements at the top of this document](#requirements-on-your-server).
-2. Verify which version you were running before and take a look at [migrations and breaking changes](../guides/migrations-and-breaking-changes.md) to see if additional steps, or configuration changes are necessary!
-3. Fully stop your old HedgeDoc server.
-4. `git pull` or unzip a new release in the directory.
+1. Check if you still meet the [requirements at the top of this document](#requirements-on-your-server).
+2. Ensure you read the [release notes](https://hedgedoc.org/releases/) of all versions between your current version
+   and the latest release.
+2. Fully stop your old HedgeDoc server.
+3. [Download](https://hedgedoc.org/latest-release/) the new release and extract it over the old directory.  
+   <small>If you use Git, you can checkout the new tag with e.g. `git checkout 1.7.2`</small>
```
Comment (ErikMichelson) [inline docs/content/setup/manual-setup.md:61]: People probably need to fetch the repo updates first

```suggestion
   <small>If you use Git, you can checkout the new tag with e.g. `git fetch origin && git checkout 1.7.2`</small>
```

File: docs/content/setup/manual-setup.md [hunk 3]
```diff
@@ -1,41 +1,139 @@
...
 5. Run `bin/setup`. This will take care of installing dependencies. It is safe to run on an existing installation.
-6. Build front-end bundle by `yarn run build` (use `yarn run dev` if you are in development).
-7. It is recommended to start your server manually once: `NODE_ENV=production yarn start`, this way it's easier to see warnings or errors that might occur (leave out `NODE_ENV=production` for development).
+6. *:octicons-light-bulb-16: If you used the release tarball for 1.7.0 or newer, this step can be skipped.*  
+   Build the frontend bundle by running `yarn run build`.  
+   <small>For debugging purposes you can instead build an unoptimized bundle with source-maps using `yarn run dev`</small>
```
Comment (ErikMichelson) [inline docs/content/setup/manual-setup.md:65]: Same as my comment above about  `yarn run dev`

File: docs/content/setup/manual-setup.md [hunk 4]
```diff
@@ -1,41 +1,139 @@
...
+# After=mysql.service
+# Uncomment if you use PostgreSQL
+# After=postgresql.service
+
+[Service]
+Type=exec
+Environment=NODE_ENV=production
+Restart=always
+RestartSec=2s
+ExecStart=/usr/bin/npm start --production
```
Comment (ErikMichelson) [inline docs/content/setup/manual-setup.md:106]: Why do we use npm here and not yarn?

File: docs/content/setup/getting-started.md [hunk 5]
```diff
@@ -6,5 +6,28 @@ To set up your instance follow these steps:
...
+
+!!! warning
+    Before you upgrade, **always read the release notes**.  
+    You can find them on our [releases page](https://hedgedoc.org/releases/).
+
+## Migrating from CodiMD & HackMD
+Migrating from CodiMD <= 1.6.0 or HackMD <= 1.1.0 to HedgeDoc should be safe,
+just make sure to read the release notes.  
+A particular issue that has come up is when handling TLS connections when using a reverse proxy.
+You must [set the `X-Forwarded-Proto` header correctly](https://docs.hedgedoc.org/guides/reverse-proxy/).
```
Comment (DerMolly) [inline docs/content/setup/getting-started.md:29]: I think, it's best if the user updated to CodiMD 1.6.0 before making the switch… And do you mean HackMDs CodiMD or an super early version of HackMDs CodiMD still named HackMD?

File: docs/content/setup/manual-setup.md [hunk 6]
```diff
@@ -1,41 +1,139 @@
...
+      The database must use charset `utf8`. This is typically the default in PostgreSQL and SQLite.  
+      In MySQL and MariaDB UTF-8 might need to be set with `alter database <DBNAME> character set utf8 collate utf8_bin;`  
+      Be aware of older MySQL and MariaDB versions which sometimes use shorter representations of UTF-8 than 4 bytes.
+      This can break if symbols with more bytes are used.
+      You can use `alter database <DBNAME> character set utf8mb4 COLLATE utf8mb4_unicode_ci` to be on the safe side.
+    - NPM (and its dependencies, [node-gyp](https://github.com/nodejs/node-gyp#installation))
+    - [Yarn Classic](https://classic.yarnpkg.com) 1.22 or higher (Yarn 2 is currently not supported)
+    - Bash (for the setup script)
+    - For **building** the HedgeDoc frontend you need a machine with at least **2 GB** RAM.
+      - Starting with release 1.7 the release tarball includes the frontend, so building it yourself is not necessary.
```
Comment (DerMolly) [inline docs/content/setup/manual-setup.md:15]:
```suggestion
      - Starting with release 1.7 the release tarball includes the prebuilt frontend, so building it yourself is not necessary.
```

File: docs/content/setup/manual-setup.md [hunk 7]
```diff
@@ -1,41 +1,139 @@
...
+
+!!! warning
+    - In this example, you must configure HedgeDoc using the `config.json` file and the 
+    `production` key.
+    - Make sure the user and group `hedgedoc` exists and has appropriate permissions in the
+    directory you installed HedgeDoc in or change the `User` and `Group` settings in the unit
+    file.
+    - Make sure `WorkingDirectory` points to the directory you installed HedgeDoc in.
+    - Make sure `ReadWritePaths` contains all directories HedgeDoc might write to. This may
+    include the `public/uploads` folder if you configure local storage. If you use SQLite, you
```
Comment (DerMolly) [inline docs/content/setup/manual-setup.md:85]:
```suggestion
    include the `public/uploads` folder if you configured local storage. If you use SQLite, you
```

File: docs/content/setup/docker.md [hunk 8]
```diff
@@ -1,20 +1,91 @@
...
 
-## Alpine-based version
+The easiest way to get started with HedgeDoc and Docker is using the following `docker-compose.yml`:
 
-[![Docker Repository on Quay](https://quay.io/repository/hedgedoc/hedgedoc/status "Docker Repository on Quay")](https://quay.io/repository/hedgedoc/hedgedoc)
+!!! warning
+    This is a minimal example to get started quickly and not intended for production use.
 
-The easiest way to setup HedgeDoc using docker are using the following three commands:
+```yaml
```
Comment (DerMolly) [inline docs/content/setup/docker.md:17]: While trying this, I could not upload an image…

Comment (ErikMichelson) [reply docs/content/setup/docker.md:17]: That's the reason why the original compose file had imgur by default.
Using local file uploads by default requires setting `CMD_DOMAIN`.

File: docs/content/setup/getting-started.md (hunk 5 above)
Comment (davidmehren) [reply docs/content/setup/getting-started.md]: `CodiMD <= 1.6.0 or HackMD <= 1.1.0` is "our" code

File: docs/content/setup/manual-setup.md (hunk 7 above)
Comment (davidmehren) [reply docs/content/setup/manual-setup.md:82]: I believe "configure" also works here

File: docs/content/setup/docker.md (hunk 8 above)
Comment (davidmehren) [reply docs/content/setup/docker.md:17]: I added `domain` and `urlAddPort` to both example configs, so uploads immediately work on localhost

File: docs/content/setup/manual-setup.md (hunk 7 above)
Comment (DerMolly) [reply docs/content/setup/manual-setup.md:83]: I think it should carry the tense in the list point above, where you used 'installed'. But this is not the hill I want to die on.

**Review by InnayTool — APPROVED** [review]

File: docs/content/setup/docker.md (hunk 8 above)
Comment (DerMolly) [reply docs/content/setup/docker.md:17]: Can confirm: Docker works now with uploads.

**Comment by mrdrogdrog** [comment]
"WARNING -  A relative path to 'guides/migrations-and-breaking-changes.md' is included in the 'nav' configuration, which is not found in the documentation files"

File: docs/content/guides/migrations-and-breaking-changes.md [hunk 9]
```diff
@@ -1,55 +0,0 @@
-# Migrations and Notable Changes
```
Comment (mrdrogdrog) [inline docs/content/guides/migrations-and-breaking-changes.md:1]: If you remove this file then you should also remove it from the navigation in `mkdocs.yml`

File: docs/content/setup/docker.md [hunk 10]
```diff
@@ -1,20 +1,93 @@
 # HedgeDoc Docker Image
 
-[![Try in PWD](https://cdn.rawgit.com/play-with-docker/stacks/cff22438/assets/images/button.png)](http://play-with-docker.com?stack=https://github.com/hedgedoc/container/raw/master/docker-compose.yml&stack_name=hedgedoc)
+!!! info "Requirements on your server"
+    - [Git](https://git-scm.com/)
+    - [Docker](https://docs.docker.com/get-docker/) 17.03.1 or higher
+    - [Docker Compose](https://docs.docker.com/compose/install/)
 
-## Debian-based version
+Official docker images are [available on quay.io](https://quay.io/repository/hedgedoc/hedgedoc).
```
Comment (mrdrogdrog) [inline docs/content/setup/docker.md:8]:
```suggestion
The official docker images are [available on quay.io](https://quay.io/repository/hedgedoc/hedgedoc).
```

or 

```suggestion
Our official docker images are [available on quay.io](https://quay.io/repository/hedgedoc/hedgedoc).
```

File: docs/content/setup/docker.md [hunk 11]
```diff
@@ -1,20 +1,93 @@
...
+We currently only support the `amd64` architecture.
 
-[![Docker Repository on Quay](https://quay.io/repository/hedgedoc/hedgedoc/status "Docker Repository on Quay")](https://quay.io/repository/hedgedoc/hedgedoc)
 
-## Alpine-based version
+The easiest way to get started with HedgeDoc and Docker is using the following `docker-compose.yml`:
```
Comment (mrdrogdrog) [inline docs/content/setup/docker.md:12]:
```suggestion
The easiest way to get started with HedgeDoc and Docker is to use the following `docker-compose.yml`:
```

File: docs/content/setup/docker.md [hunk 12]
```diff
@@ -1,20 +1,93 @@
...
+
+    In order to migrate the existing database to the new default credentials, run
+    ```shell
+    docker-compose exec database createuser --superuser -Uhackmd postgres
+    docker-compose exec database psql -Upostgres -c "alter role hackmd rename to hedgedoc; alter role hedgedoc with password 'password'; alter database hackmd rename to hedgedoc;"
+    ```
+    before running `docker-compose up`.
+
+You can upgrade to the latest release by stopping the containers and changing the referenced image version in `docker-compose.yml`.  
+Then run `docker-compose up` to start HedgeDoc again.
```
Comment (mrdrogdrog) [inline docs/content/setup/docker.md:69]: Maybe `docker-compose up -d`?

File: docs/content/setup/docker.md [hunk 13]
```diff
@@ -1,20 +1,93 @@
...
+
+### Migrating from CodiMD & HackMD
+
+If you currently use CodiMD or HackMD, you should be able to swap the docker image for ours.
+See [the general migration hints](/setup/getting-started/#migrating-from-codimd-hackmd) for compatibility details.
+
+
+## Backup
+
+Start your docker and enter the terminal, follow below commands:
```
Comment (mrdrogdrog) [inline docs/content/setup/docker.md:79]: `Start your docker`?
This sentences seems odd.

File: docs/content/setup/docker.md [hunk 14]
```diff
@@ -1,20 +1,93 @@
...
+
+```shell
+ docker-compose exec database pg_dump hedgedoc -U hedgedoc > backup.sql
+```
+
+
+## Restore
```
Comment (mrdrogdrog) [inline docs/content/setup/docker.md:86]:
```suggestion
## Restore a backup
```

File: docs/content/setup/docker.md (hunk 13 above)
Comment (mrdrogdrog) [inline docs/content/setup/docker.md:79]: What about the volumes / directories to backup?

File: docs/content/setup/getting-started.md [hunk 15]
```diff
@@ -6,5 +6,28 @@ To set up your instance follow these steps:
...
 3. [Configure HedgeDoc](https://docs.hedgedoc.org/configuration/)
+4. If you didn't disable [local accounts](/configuration/#email-local-account), you can use the "Sign In" button to
+   create an account, login and start using HedgeDoc.
 
 Follow us on <a href="https://social.hedgedoc.org/" target="_blank" rel="noreferer noopener">:fontawesome-brands-mastodon:{: .mastodon }Mastodon</a> or <a href="https://social.hedgedoc.org/twitter" target="_blank" rel="noreferer noopener">:fontawesome-brands-twitter:{: .twitter }Twitter</a> for updates.
+
+## Upgrading HedgeDoc
+
+HedgeDoc follows [Semantic Versioning](https://semver.org/).  
+This means that minor and patch releases should not introduce user-facing backwards-incompatible changes.
```
Comment (mrdrogdrog) [inline docs/content/setup/getting-started.md:17]:
```suggestion
This means that minor and patch releases shouldn't introduce user-facing backwards-incompatible changes.
```

File: docs/content/setup/getting-started.md [hunk 16]
```diff
@@ -6,5 +6,28 @@ To set up your instance follow these steps:
...
+
+!!! warning
+    Before you upgrade, **always read the release notes**.  
+    You can find them on our [releases page](https://hedgedoc.org/releases/).
+
+## Migrating from CodiMD & HackMD
+Migrating from CodiMD <= 1.6.0 or HackMD <= 1.1.0 to HedgeDoc should be safe,
+just make sure to read the release notes.  
+A particular issue that has come up is when handling TLS connections using a reverse proxy.
+You must [set the `X-Forwarded-Proto` header correctly](https://docs.hedgedoc.org/guides/reverse-proxy/#reverse-proxy-config).
```
Comment (mrdrogdrog) [inline docs/content/setup/getting-started.md:29]: Random information?

File: docs/content/setup/manual-setup.md [hunk 17]
```diff
@@ -1,41 +1,137 @@
...
+    - NPM (and its dependencies, [node-gyp](https://github.com/nodejs/node-gyp#installation))
+    - [Yarn Classic](https://classic.yarnpkg.com) 1.22 or higher (Yarn 2 is currently not supported)
+    - Bash (for the setup script)
+    - For **building** the HedgeDoc frontend you need a machine with at least **2 GB** RAM.
+      - Starting with release 1.7 the release tarball includes the prebuilt frontend, so building it yourself is not necessary.
+
+1. Check if you meet the [requirements at the top of this document](#manual-installation).
+2. Download the [latest release](https://hedgedoc.org/latest-release/) and extract it.  
+   <small>Alternatively, you can use Git to clone the repository and checkout a release, e.g. with `git clone -b 1.7.2 https://github.com/hedgedoc/hedgedoc.git`.</small>
+3. Enter the directory and type `bin/setup`, which will install the dependencies and create example configs.
```
Comment (mrdrogdrog) [inline docs/content/setup/manual-setup.md:20]:
```suggestion
3. Enter the directory and execute `bin/setup`, which will install the dependencies and create example configs.
```

File: docs/content/setup/manual-setup.md [hunk 18]
```diff
@@ -1,41 +1,137 @@
...
+         "storage": "./db.hedgedoc.sqlite"
+      },
+       "urlAddPort": true,
+       "domain": "localhost"
+     }
+   }
+   ```
+   It's also possible to use environment variables.
+   For details, have a look at [the configuration documentation](../configuration.md).
+5. *:octicons-light-bulb-16: If you used the release tarball for 1.7.0 or newer, this step can be skipped.*
```
Comment (mrdrogdrog) [inline docs/content/setup/manual-setup.md:36]: Is there a particular reason why you chose this icon?

Comment (mrdrogdrog) [inline docs/content/setup/manual-setup.md:36]:
```suggestion
5. *:octicons-light-bulb-16: If you use the release tarball for 1.7.0 or newer, this step can be skipped.*  
```

File: docs/content/setup/manual-setup.md [hunk 19]
```diff
@@ -1,41 +1,137 @@
...
+   For details, have a look at [the configuration documentation](../configuration.md).
+5. *:octicons-light-bulb-16: If you used the release tarball for 1.7.0 or newer, this step can be skipped.*  
+   Build the frontend bundle by running `yarn run build`.
+6. It is recommended to start your server manually once:  
+   ```shell
+   NODE_ENV=production yarn start
+   ```
+   This way it's easier to see warnings or errors that might occur.  
+   <small>You can leave out `NODE_ENV=production` for development.</small>  
+7. If you used the example config, HedgeDoc should now be available at [http://127.0.0.1:3000](http://127.0.0.1:3000).
```
Comment (mrdrogdrog) [inline docs/content/setup/manual-setup.md:44]:
```suggestion
7. If you use the example config, HedgeDoc should now be available at [http://127.0.0.1:3000](http://127.0.0.1:3000).
```

File: docs/content/setup/manual-setup.md [hunk 20]
```diff
@@ -1,41 +1,137 @@
...
+8. Run the server as you like (node, forever, pm2, systemd, Init-Scripts).  
+   See [below](#systemd-unit-example) for an example using systemd.
+
+## Upgrading
+
+!!! warning
+    Before you upgrade, **always read the release notes**.  
+    You can find them on our [releases page](https://hedgedoc.org/releases/).
 
 If you are upgrading HedgeDoc from an older version, follow these steps:
```
Comment (mrdrogdrog) [inline docs/content/setup/manual-setup.md:54]:
```suggestion
If you upgrade HedgeDoc from an older version, follow these steps:
```

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

File: docs/content/setup/docker.md (hunk 13 above)
Comment (DerMolly) [reply docs/content/setup/docker.md:79]: `Start your docker daemon`, maybe?

File: docs/content/setup/docker.md (hunk 10 above)
Comment (mrdrogdrog) [reply docs/content/setup/docker.md:8]: imho this extra word makes the sentence more complete.

File: docs/content/setup/docker.md (hunk 11 above)
Comment (mrdrogdrog) [reply docs/content/setup/docker.md:12]: personal preference :shrug:

File: docs/content/setup/getting-started.md (hunk 15 above)
Comment (mrdrogdrog) [reply docs/content/setup/getting-started.md:17]: personal preference :shrug:

File: docs/content/setup/docker.md (hunk 12 above)
Comment (davidmehren) [reply docs/content/setup/docker.md:69]: Hm, we use the undaemonized version before in the install section and in line 66.
I believe everyone who wants to use `-d` will be able do add it themselves.

File: docs/content/setup/manual-setup.md (hunk 18 above)
Comment (mrdrogdrog) [reply docs/content/setup/manual-setup.md:36]: IMHO if you use present tense in every other bullet point and you should use it everywhere. :shrug:

File: docs/content/setup/manual-setup.md (hunk 19 above)
Comment (mrdrogdrog) [reply docs/content/setup/manual-setup.md:44]: same as above

File: docs/content/setup/manual-setup.md (hunk 20 above)
Comment (mrdrogdrog) [reply docs/content/setup/manual-setup.md:54]: same as above

File: docs/content/setup/docker.md (hunk 13 above)
Comment (davidmehren) [reply docs/content/setup/docker.md:79]: Currently not in scope, as I don't want to spend *even more* time on this PR

File: docs/content/setup/getting-started.md (hunk 16 above)
Comment (davidmehren) [reply docs/content/setup/getting-started.md:29]: Nah, we had that issue come up so many times that it seems sensible to highlight it

File: docs/content/setup/manual-setup.md (hunk 18 above)
Comment (davidmehren) [reply docs/content/setup/manual-setup.md:36]: It is a helpful tip, a lightbulb lights the way? xD

File: docs/content/setup/docker.md (hunk 13 above)
Comment (davidmehren) [reply docs/content/setup/docker.md]: I rephrased both the backup and the restore section

File: docs/content/setup/manual-setup.md (hunk 20 above)
Comment (davidmehren) [reply docs/content/setup/manual-setup.md]: Worded it a bit defferently now

**Review by mrdrogdrog — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Pull request merged**